assets/scene_renderer/shaders/Config.hlsli

```diff
@@ -0,0 +1,149 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONFIG_HLSLI
+#define CONFIG_HLSLI
+
+#if defined(__spirv__)
+#define DECLARE_LOCATION(LOC_NUM) [[vk::location(LOC_NUM)]]
+#else
+#define DECLARE_LOCATION(LOC_NUM)
+#endif
+
+#define INVALID_INDEX 0xFFFFFFFF
+
+#define MAX_UNIQUE_MATERIALS      8192
+#define MAX_TEXTURES_PER_MATERIAL 6
+#define MAX_MATERIAL_TEXTURES     (MAX_UNIQUE_MATERIALS * MAX_TEXTURES_PER_MATERIAL)
+
+#define MAX_MATERIAL_SAMPLERS     64
```

Comment:
@chaoticbob can you elaborate why these limits? i'm not finding docs pointing to them.

the max unique material count looks something like the max dimensions you'd want to support on textures, i'm not sure i understand why we're using that as the limit for material count

Reply:
There's no set convention for the number of samplers. It can be whatever the program wants it to be based on its own use cases. 64 was just arbitarily selected knowing that at at least 6 would be used for the PBR rendering with the remainder being availabe in case there's more specialization needed. 64 is also reasonable number to have given the variance across [maxPerStageDescriptorSamplers](https://vulkan.gpuinfo.org/displaydevicelimit.php?platform=windows&name=maxPerStageDescriptorSamplers), even if it seems a little low.

The doc that @Keenuts specifies the tier limits, much akin to Vulkan's physical device limits. You don't need to max them out unless there's some reason to do so.

Reply:
@Keenuts going to mark this as resolved, seems this limit is based on judgement call for what we expect to be a safe value for now